meshtastic/interfaces/ble/lifecycle_receive_runtime.py

```
@@ -0,0 +1,397 @@
+
+        Treats ThreadLike proxies with the same thread ident as the current
+        receive thread as equivalent to the current thread.
+        """
+        if thread is threading.current_thread():
+            return True
+        if thread is None:
+            return False
+        thread_ident, _ = _thread_start_probe(thread)
+        return isinstance(thread_ident, int) and thread_ident == threading.get_ident()
+
+    def _check_receive_start_conditions(
```

Ruff (0.15.6)

[warning] Too many return statements (10 > 6)

(PLR0911)